apps/api/src/controllers/competition.controller.ts

```diff
@@ -485,6 +511,23 @@ export function makeCompetitionController(services: ServiceRegistry) {
           ? CompetitionStatusSchema.parse(req.query.status)
           : undefined;
         const pagingParams = PagingParamsSchema.parse(req.query);
+        // Only cache frontend requests (no user or agent context)
+        const isPublicRequest = checkIsPublicRequest(req);
+
+        // Build a cache key off stable query params for public requests, only if cache is enabled
+        const isCacheEnabled = checkIsCacheEnabled();
```

Comment:
for tests to pass, i disabled the caching (and also disable caching for development mode)